src/btree/bt_discard.c

```diff
@@ -33,7 +33,7 @@ __wt_ref_out(WT_SESSION_IMPL *session, WT_REF *ref)
 
     /*
      * Make sure no other thread has a hazard pointer on the page we are about to discard. This is
-     * complicated by the fact that readers publish their hazard pointer before re-checking the page
+     * complicated by the fact that readers share their hazard pointer before re-checking the page
```

Comment:
Based on the conversation on Slack is this is one of the uses of "publish" that we can keep? I prefer publish over share here. I'll hold off on further review until we've finished updating all the comments

Reply:
So we'll keep publish as WiredTigers way of saying write to a shared structure?